features/environment.py

```diff
@@ -98,36 +98,57 @@ def from_environ(cls) -> "UAClientBehaveConfig":
 
 
 def before_all(context: Context) -> None:
-    """behave will invoke this before anything else happens.
-
-    In this function, we launch a container, install ubuntu-advantage-tools and
-    then capture an image.  This image is then reused by each feature, reducing
-    test execution time.
-    """
+    """behave will invoke this before anything else happens."""
     userdata = context.config.userdata
     context.reuse_container = userdata.get("reuse_container")
+    context.series = {}
+    context.image_name = ""
```

Comment:
The way you have structured the test, context.series will overwrite image_name on each test run. This yields warnings during test runs because you keep overwriting that context.image_name value. 

I think it might be better if we drop context.image_name and rename context.series to ``context.series_image_names``.

```suggestion
    context.series_image_names = {}
```

Reply:
Ok.